### PR TITLE
Make supervisor a server and introduce modyn client

### PR DESCRIPTION
**Pull request by jenny011**
WIP
Addresses issues #302, #317

### CODE REVIEW

File: modyn/supervisor/internal/pipeline_executor/pipeline_executor.py [hunk 1]
```diff
@@ -437,14 +284,10 @@ def _handle_triggers_within_batch(self, batch: list[tuple[int, int, int]], trigg
...
 
-            num_samples_in_trigger = self.grpc.get_number_of_samples(self.pipeline_id, trigger_id)
-            if num_samples_in_trigger > 0:
-                self.status_bar.update(demo="Training")
-                self._run_training(trigger_id)  # Blocks until training is done.
-            else:
-                logger.info(f"Skipping training on empty trigger {trigger_id}]")
-
-            self.status_bar.update(demo="Handling triggers")
+            self._run_training(trigger_id)  # Blocks until training is done.
```
Comment (MaxiBoether) [inline modyn/supervisor/internal/pipeline_executor/pipeline_executor.py:287]: https://github.com/eth-easl/modyn/commit/2366cb494372f2672ecd186906b9bffe7d3843c8 got lost during the merge

File: tmuxp.yaml
Comment (MaxiBoether) [inline tmuxp.yaml:1]: I am commenting on this file but it does affect another:

Could you update docs/EXAMPLE.md, in particular the subsection on where we describe how to run the example pipeline, to explain how to use the new CLI instead? Thank you!

**Review by MaxiBoether — CHANGES_REQUESTED** [review]
Thank you so much! I went through everything again and it looks very clean. I had two comments, one documentation thing and one commit that got lost during the merge I think. After fixing that, we can finally merge into main. Thanks for your hard work!

File: modyn/supervisor/internal/pipeline_executor/pipeline_executor.py (hunk 1 above)
Comment (jenny011) [reply modyn/supervisor/internal/pipeline_executor/pipeline_executor.py]: fixed

File: tmuxp.yaml
Comment (jenny011) [reply tmuxp.yaml:1]: Updated EXAMPLE.md, supervisor/README.md, modynclient READMEs.

**Review by MaxiBoether — APPROVED** [review]
Thank you!

**Pull request merged**